tests/atoms/gaia_apps.js

```diff
@@ -280,24 +270,17 @@ var GaiaApps = {
 
   launch: function(app, appName, launchPath, entryPoint) {
     if (app) {
+      let manifestURL = app.manifestURL;
       let origin = app.origin;
 
       let sendResponse = function() {
-        let appWindow = GaiaApps.getAppByURL(app.origin + launchPath);
-        let origin = appWindow.origin;
-        let result = {
-          frame: (appWindow.browser) ?
-              appWindow.browser.element :
-              appWindow.frame.firstChild,
-          src: (appWindow.browser) ?
-              appWindow.browser.element.src :
-              appWindow.iframe.src,
-          name: appWindow.name,
-          origin: origin};
+        let result = GaiaApps.getDisplayedApp();
         marionetteScriptFinished(result);
       };
 
-      if (GaiaApps.getDisplayedApp().origin == origin) {
+      let displayedApp = GaiaApps.getDisplayedApp();
+      if (displayedApp.manifestURL == manifestURL ||
+          (!displayedApp.manifestURL && displayedApp.origin == origin)) {
         console.log('app with origin \'' + origin + '\' is already running');
```

Comment:
We compare manifestURL, probably need to log more than origin since it's secondary condition.